ctx_test.go

```diff
@@ -690,4 +690,147 @@ func TestNetHttpContext_Params(t *testing.T) {
 		assert.Equal(t, uint64(8923), params.MyHeader)
 		assert.InEpsilon(t, float32(20.30), params.Temperature, 0.01)
 	})
+
+	t.Run("support for array of strings", func(t *testing.T) {
+		type MyParams struct {
+			Tags []string `query:"tags"`
+		}
+
+		r := httptest.NewRequest("GET", "http://example.com/foo?tags=golang&tags=web&tags=api", nil)
+		w := httptest.NewRecorder()
+		c := NewNetHTTPContext[any, MyParams](BaseRoute{}, w, r, readOptions{})
+		params, err := c.Params()
+		require.NoError(t, err)
+		require.NotEmpty(t, params)
+		assert.Equal(t, []string{"golang", "web", "api"}, params.Tags)
+	})
+
+	t.Run("support for array of integers", func(t *testing.T) {
+		type MyParams struct {
+			IDs []int `query:"ids"`
+		}
+
+		r := httptest.NewRequest("GET", "http://example.com/foo?ids=1&ids=2&ids=3", nil)
+		w := httptest.NewRecorder()
+		c := NewNetHTTPContext[any, MyParams](BaseRoute{}, w, r, readOptions{})
+		params, err := c.Params()
+		require.NoError(t, err)
+		require.NotEmpty(t, params)
+		assert.Equal(t, []int{1, 2, 3}, params.IDs)
+	})
+
+	t.Run("support for array of various integer types", func(t *testing.T) {
+		type MyParams struct {
+			Int8s   []int8   `query:"int8s"`
+			Int16s  []int16  `query:"int16s"`
+			Int32s  []int32  `query:"int32s"`
+			Int64s  []int64  `query:"int64s"`
+			Uints   []uint   `query:"uints"`
+			Uint8s  []uint8  `query:"uint8s"`
+			Uint16s []uint16 `query:"uint16s"`
+			Uint32s []uint32 `query:"uint32s"`
+			Uint64s []uint64 `query:"uint64s"`
+		}
+
+		url := "http://example.com/foo?" +
+			"int8s=1&int8s=2&" +
```

Comment:
Please also tests how the code reacts to overflow, like settings 1024 into an int8 that cannot go further than 128